### ISSUE TITLE
Add named-grid-areas-no-invalid

### ISSUE THREAD

**Issue by evrom** [stylelint/stylelint#5155]
> What is the problem you're trying to solve?

Validating `grid-template-areas` property values. 

> What solution would you like to see?

A rule for `grid-template-areas`

I  made a `grid-template-areas` validator. The code is here: https://github.com/evrom/grid-validator and a live version is here: https://gridvalidator.com/ . I can convert the code into a rule, but I want to make sure it is within the scope of features that stylelint wants in their project.

**Comment by jeddy3** [comment]
@evrom Thanks for the request and for using the template.

That's a fascinating validator. 

I believe it's [within the scope](https://stylelint.io/user-guide/rules/about#about-rules) of stylelint built-in rules as we validate simple cases (and offload complex syntax validation to things like CSSTree).

From the [spec](https://drafts.csswg.org/css-grid/#grid-template-areas-property):

> All strings must define the same number of cell tokens (named cell tokens and/or null cell tokens), and at least one cell token, or else the declaration is invalid. If a named grid area spans multiple grid cells, but those cells do not form a single filled-in rectangle, the declaration is invalid.

- Name: `named-grid-areas-no-invalid` (where "named-grid-areas" is the _thing_ and "no-invalid" is the _check_)
- Primary option: `true`
- Secondary options: None
- Autofixable: No
- Message:
   - Unexpected string without cell token
   - Unexpected non-rectangle
- Description: "Disallow invalid named grid areas."
- Section: "Possible errors" -> "Named grid areas"

I'm not 100% sure of the messages, yet. We can flesh those out once the rule has taken shape in a pull request.

We'll want to use the value parser to identify the string nodes so that we avoid issues with comments, e.g.:

```css
a {
  grid-template-areas: 
    /* "" */ "a a ."
    "a a ."
    ". b c";
}
```

As a "possible errors" rule, it's a candidate for the recommended config.

I've labelled the issue as ready to implement. @evrom Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide.
